src/OrleansRuntime/Core/InsideRuntimeClient.cs

```diff
@@ -336,6 +341,15 @@ public void SniffIncomingMessage(Message message)
                 try
                 {
                     var request = (InvokeMethodRequest) message.BodyObject;
+                    if (request.Arguments != null)
+                    {
+                        for (var i = 0; i < request.Arguments.Length; i++)
+                        {
+                            var arg = request.Arguments[i];
+                            if (!(arg is GrainCancellationToken)) continue;
```

Comment:
Maybe `UnwrapCancellationToken` should be renamed to `RegisterCancellationToken` or something along those lines, since it no longer unwraps them.